src/Castle.Facilities.Quartz/Scheduler.cs

```diff
@@ -1063,129 +1155,24 @@ public IJobFactory JobFactory
-            await Stop(token);
-        }
+            var properties = new NameValueCollection();
 
-        #region Methods that are here for backwards-compatibility (maybe they aren't needed anymore)
-
-        /// <summary>
-        ///     Set by Castle via de configuration
-        /// </summary>
-        /// <remarks>This method is added for the backwards-compatibility with Quartz v1 and the QuartzFacility v1</remarks>
-        public IJobListener[] SetGlobalJobListeners
-        {
-            set
+            // Set properties
+            if (props != null)
             {
-                if (value != null)
-                    foreach (var jobListener in value)
-                        _scheduler.ListenerManager.AddJobListener(jobListener);
+                foreach (var prop in props.Keys)
+                    properties[prop] = props[prop];
             }
-        }
 
-        /// <summary>
-        ///     Set by Castle via de configuration
-        /// </summary>
-        /// <remarks>This method is added for the backwards-compatibility with Quartz v1 and the QuartzFacility v1</remarks>
-        public ITriggerListener[] SetGlobalTriggerListeners
-        {
-            set
-            {
-                if (value == null) return;
-
-                foreach (var triggerListener in value)
-                    _scheduler.ListenerManager.AddTriggerListener(triggerListener);
-            }
-        }
-
-        /// <summary>
-        ///     Set by Castle via de configuration
-        /// </summary>
-        /// <remarks>This method is added for the backwards-compatibility with Quartz v1 and the QuartzFacility v1</remarks>
-        public IDictionary JobListeners
-        {
-            set
-            {
-                if (value == null) return;
-
-                foreach (DictionaryEntry jobListenerDictionaryEntry in value)
-                {
-                    if (!(jobListenerDictionaryEntry.Value is IList)) continue;
-
-                    foreach (IJobListener jobListener in (IList)jobListenerDictionaryEntry.Value)
-                        _scheduler.ListenerManager.AddJobListener(jobListener);
-                }
-            }
-        }
-
-        /// <summary>
-        ///     Set by Castle via de configuration
-        /// </summary>
-        /// <remarks>This method is added for the backwards-compatibility with Quartz v1 and the QuartzFacility v1</remarks>
-        public IDictionary TriggerListeners
-        {
-            set
-            {
-                if (value == null) return;
-
-                foreach (DictionaryEntry triggerListenerDictionaryEntry in value)
-                {
-                    if (!(triggerListenerDictionaryEntry.Value is IList)) continue;
+            // Create scheduler
+            var factory = properties.Count == 0 ? new StdSchedulerFactory() : new StdSchedulerFactory(properties);
+            var getSchedulerTask = factory.GetScheduler();
+            getSchedulerTask.Wait();
```

Comment:
You are calling `Result` below, `Wait()` is unnecessary.

Reply:
When synchronously waiting for tasks, `task.GetAwaiter().GetResult()` is the way to go here. This has the equivalent effect of `.ConfigureAwait(false)` _and_ it unwraps exceptions instead of throwing `AggregateException`.

Reply:
replaced by factory.GetScheduler().GetAwaiter().GetResult()